controllers/sink.go

```diff
@@ -49,33 +45,33 @@ func (r *SinkReconciler) ObserveSinkStatefulSet(ctx context.Context, req ctrl.Re
 	}, statefulSet)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			r.Log.Info("sink is not ready yet...")
-			return nil
+			r.Log.Info("sink statefulset is not found...")
+		} else {
+			sink.Status.Conditions[v1alpha1.StatefulSet] = condition
+			return err
 		}
-		return err
+	} else {
+		// statefulset created, waiting it to be ready
+		condition.Action = v1alpha1.Wait
 	}
 
 	selector, err := metav1.LabelSelectorAsSelector(statefulSet.Spec.Selector)
 	if err != nil {
 		r.Log.Error(err, "error retrieving statefulSet selector")
-		return err
+	} else {
```

Comment:
this `else` clause is not necessary. 

I think golang has a error check pattern that only do `if err != nil`, otherwise the program just proceed normally.

Reply:
if err != nil mostly selector will be nil, so I think `else` is needed to avoid NPE